From 02d9fbbd1cd61ba9ee7a71fc566c65a674ad1a70 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Erdmann?= <dywi@mailerd.de>
Date: Sun, 6 Jan 2013 23:26:18 +0100
Subject: [PATCH] tlp-usblist: print <unknown> if device $desc is empty

---
 tlp-usblist | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tlp-usblist b/tlp-usblist
index 039b4ae..260b655 100755
--- a/tlp-usblist
+++ b/tlp-usblist
@@ -75,8 +75,9 @@ foreach $udev (grep { ! /:/ } glob "/sys/bus/usb/devices/*") {
 
 # Output device list with attributes and drivers
 foreach (`lsusb`) {
-    my ($bus, $dev, $usbid, $desc) = /Bus (\S+) Device (\S+): ID (\S+) (.*)/;
+    my ($bus, $dev, $usbid, $desc) = /Bus (\S+) Device (\S+): ID (\S+)[ ]+(.*)/;
     my $usbk = $bus . "_" . $dev;
+    $desc ||= "<unknown>";
     print "Bus $bus Device $dev ID $usbid $usbdevices{$usbk}[1] -- $desc (" 
         . usbdriverlist ($usbdevices{$usbk}[0]) . ")\n";
 }
-- 
1.8.0.2

